### ISSUE TITLE
DatabaseManager' is not assignable to type v1.21.1

### ISSUE THREAD

**Issue by lalit774** [backstage/backstage#22108]
### 📜 Description

Type error databaseManager

 const databaseManager = DatabaseManager.fromConfig(config, {
    logger: root,
  });
const taskScheduler = TaskScheduler.fromConfig(config, { databaseManager });

### 👍 Expected behavior

we should not get type issue.
if we use following in packages/backend/src/index.ts. it works fine.
 const databaseManager:any = DatabaseManager.fromConfig(config, {
    logger: root,
  });

### 👎 Actual Behavior with Screenshots

Types have separate declarations of a private property 'config'.ts(2322)
(property) databaseManager?: DatabaseManager | undefined

### 👟 Reproduction steps

backstage version 1.20.0
1) Create a new backstage app 
2 yarn tsc
3) you will see typescript warning

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by camilaibs** [comment]
Hi @lalit774, thank you for opening an issue 🙏🏻 .
Could you run `npx @backstage/create-app@latest` at the latest version and tell us whether you get the same issue or not?

**Comment by tonglil** [comment]
I'm running into this issue where building with the multistage docker file fails, but `yarn tsc` locally outside of docker runs fine.

```
yarn tsc
yarn run v1.22.21
$ tsc
✨  Done in 1.96s.
```
```
docker image build -t backstage-multistage .
[+] Building 7.9s (18/24)                                                                                                                                               docker:desktop-linux
 => [internal] load build definition from Dockerfile                                                                                                                                    0.0s
 => => transferring dockerfile: 3.88kB                                                                                                                                                  0.0s
 => [internal] load metadata for docker.io/library/node:18-bookworm-slim                                                                                                                1.3s
 => [internal] load .dockerignore                                                                                                                                                       0.0s
 => => transferring context: 141B                                                                                                                                                       0.0s
 => [packages 1/6] FROM docker.io/library/node:18-bookworm-slim@sha256:27c468a7089b35374ec79b2a49ae9911818e37665caffe4d4d531b74c78cfbe3                                                 0.0s
 => [internal] load build context                                                                                                                                                       0.0s
 => => transferring context: 30.88kB                                                                                                                                                    0.0s
 => CACHED [internal] settings cache mount permissions                                                                                                                                  0.0s
 => CACHED [build  2/10] RUN --mount=type=cache,target=/var/cache/apt,sharing=locked     --mount=type=cache,target=/var/lib/apt,sharing=locked     apt-get update &&     apt-get insta  0.0s
 => CACHED [build  3/10] RUN --mount=type=cache,target=/var/cache/apt,sharing=locked     --mount=type=cache,target=/var/lib/apt,sharing=locked     apt-get update &&     apt-get insta  0.0s
 => CACHED [build  4/10] WORKDIR /app                                                                                                                                                   0.0s
 => CACHED [packages 2/6] WORKDIR /app                                                                                                                                                  0.0s
 => CACHED [packages 3/6] COPY package.json yarn.lock ./                                                                                                                                0.0s
 => CACHED [packages 4/6] COPY packages packages                                                                                                                                        0.0s
 => CACHED [packages 5/6] COPY plugins plugins                                                                                                                                          0.0s
 => CACHED [packages 6/6] RUN find packages ! -name "package.json" -mindepth 2 -maxdepth 2 -exec rm -rf {} +                                                                            0.0s
 => CACHED [build  5/10] COPY --from=packages --chown=node:node /app .                                                                                                                  0.0s
 => CACHED [build  6/10] RUN --mount=type=cache,target=/home/node/.cache/yarn,sharing=locked,uid=1000,gid=1000     yarn install --frozen-lockfile --network-timeout 600000              0.0s
 => CACHED [build  7/10] COPY --chown=node:node . .                                                                                                                                     0.0s
 => ERROR [build  8/10] RUN yarn tsc                                                                                                                                                    6.5s
------                                                                                                                                                                                       
 > [build  8/10] RUN yarn tsc:                                                                                                                                                               
0.337 yarn run v1.22.19                                                                                                                                                                      
0.363 $ tsc                                                                                                                                                                                  
6.419 packages/backend/src/index.ts:43:60 - error TS2322: Type 'import("/app/packages/backend/node_modules/@backstage/backend-common/dist/index").DatabaseManager' is not assignable to type 'import("/app/node_modules/@backstage/backend-common/dist/index").DatabaseManager'.
6.419   Types have separate declarations of a private property 'config'.
6.419 
6.419 43   const taskScheduler = TaskScheduler.fromConfig(config, { databaseManager });
6.419                                                               ~~~~~~~~~~~~~~~
6.419 
6.419 
6.419 Found 1 error in packages/backend/src/index.ts:43
6.419 
6.448 error Command failed with exit code 2.
6.448 info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
------
Dockerfile:41
--------------------
  39 |     COPY --chown=node:node . .
  40 |     
  41 | >>> RUN yarn tsc
  42 |     RUN yarn --cwd packages/backend build
  43 |     # If you have not yet migrated to package roles, use the following command instead:
--------------------
ERROR: failed to solve: process "/bin/sh -c yarn tsc" did not complete successfully: exit code: 2

View build details: docker-desktop://dashboard/build/desktop-linux/desktop-linux/vggbbybcq80umf6w60kpzhfr4
```

**Comment by benjdlambert** [comment]
Looks like you have duplicates of `backend-common` in `yarn.lock` 

Could you try using the `yarn backstage-cli versions:bump` command, or by making sure that the versions are aligned by using `yarn why @backstage/backend-common`

**Comment by Rugvip** [comment]
For the immediate fix please deduplicate the installed packages as suggested by @benjdlambert, but a more permanent fix is on the way in #22835 too.
